src/core/auth/qgsauthmanager.cpp

```diff
@@ -1200,13 +1202,14 @@ bool QgsAuthManager::updateAuthenticationConfig( const QgsAuthMethodConfig &conf
 
 bool QgsAuthManager::loadAuthenticationConfig( const QString &authcfg, QgsAuthMethodConfig &mconfig, bool full )
 {
-  QMutexLocker locker( mMutex );
   if ( isDisabled() )
     return false;
 
   if ( full && !setMasterPassword( true ) )
     return false;
 
+  QMutexLocker locker( mMutex );
```

Comment:
This looks fine to me (unless we decide to move the lockers down one level), and I would say that it prevents a deadlock because setMasterPassword locks the mutext again.

Did you check if this change is enough to fix your issue?

Reply:
Yep, this PR fixes the issue I stumbled on expanding connections in the browser panel, as well as the above-pasted mini script.

Reply:
Sorry I wasn't clear, my question was if this change **alone** is sufficient to fix your issue without the need of the other change.

Reply:
@elpaso , ah, let me test this out. I doubt, but who knows.

Reply:
@elpaso , nope, it's not sufficient.